pkg/daemon/winsvc/winsvc.go

```diff
@@ -208,3 +211,23 @@ func NewTestMgr(existingServices map[string]*FakeService) *testMgr {
 	}
 	return testMgr
 }
+
+// EnsureServiceState ensures the service is in the given state
+func EnsureServiceState(service Service, state svc.State) error {
```

Comment:
we are returning err from multiple places. Would it be nice to append some additional error messages so that we will get more details while debugging?

Reply:
+1